packages/contract-wrappers/src/utils/market_orders_optimization_utils.ts

```diff
@@ -0,0 +1,44 @@
+import { SignedOrder } from '@0xproject/types';
+import * as _ from 'lodash';
+
+import { constants } from './constants';
+
+export const marketOrdersOptimizationUtils = {
+    /**
+     * Takes an array of orders and outputs an array of equivalent orders where all takerAssetData are '0x' and
+     * all makerAssetData are '0x' except for that of the first order, which retains its original value
+     * @param   orders         An array of SignedOrder objects
+     * @returns optimized orders
+     */
+    optimizeMarketOrders(orders: SignedOrder[]): SignedOrder[] {
```

Comment:
This should be `optimizeForwarderOrders`. The actual market buy and sell functions within the exchange contract will be optimized in a slightly different way.